src/main/java/bio/terra/service/dataset/DatasetDao.java

```diff
@@ -531,6 +531,16 @@ public DatasetSummary retrieveSummaryByName(String name) {
         if (!whereClauses.isEmpty()) {
             whereSql = " WHERE " + StringUtils.join(whereClauses, " AND ");
         }
+
+        // get the filtered total count of objects without offset and limit
+        String filteredTotalSql = "SELECT count(id) AS total FROM dataset WHERE " +
+            StringUtils.join(whereClauses, " AND ");
```

Comment:
Given the different calls, we should have a `@Transactional` annotation on this method (make sure that it's readonly)

Reply:
Good point!